### DISCUSSION TITLE
Docker socket is not available

### DISCUSSION THREAD

**Discussion by starworld99** [nextcloud/all-in-one#4007] [Questions] [answered]
Hi, I am new here, when I install  nextcloud/all-in-one from dockerhub on  my QNAP nas device(TS-466C), I got the following error message, how should I do? Much appreciate for you help!
Docker socket is not available. Cannot continue.
Please make sure to mount the docker socket into /var/run/docker.sock inside the container!
If you did this by purpose because you don't want the container to have access to the docker socket, see https://github.com/nextcloud/all-in-one/tree/main/manual-install.

**Comment by szaimen** [comment, marked as answer]
Hi, I guess the docker.sock is just not available under `/var/run/docker.sock` so you would need to adjust the command to mount it correctly into the container.

**Reply by starworld99** [reply]
thank you! I think I have found something, the docker on QNAP NAS is a coustom version, the docker os cannot use /var/run directly.
All dockers installation under /share/CACHEDEV1_DATA/Container/container-station-data/lib/docker.  When I add shared folder, the docker system can only browser to limited folder /share/CACHEDEV1_DATA/Container/container-station-data/lib/docker/volumes

**Reply by sbeeker** [reply]
I am not sure where this thread begins but are you trying to get docker to run in  QNAP Ubuntu Container service?  I have been all through everything looking. Right now I do not get the Socket /var/run/docker.sock. I looked at logs for snap. "snap log docker" and see a long network error message about NAT controller etc.  Your comment about docker.sock. When you install the snap docker the readme on the install says you must have everything running in $HOME.    So I am baffled.   I would be curious if anybody got this working?  Snap docker (or any docker) running on Ubuntu 22.4.2 on QNAP Container Service.
